experimental/devices/cap1188/cap1188.go

```diff
@@ -0,0 +1,554 @@
+// This is quite more efficient than looping through each led and turn them off.
+func (d *Dev) AllLedsOff() error {
+	if d.opts.LinkedLEDs {
+		return wrap(fmt.Errorf("can't manually set LEDs when they are linked to sensors"))
+	}
+	return d.regWrapper.WriteUint8(reg_LEDOutputControl, 0x00)
+}
+
+// SetLed sets the state of a LED as on or off
+// Only works if the LEDs are not linked to the sensors
+func (d *Dev) SetLed(idx int, state bool) error {
+	if d.opts.LinkedLEDs {
+		return wrap(fmt.Errorf("can't manually set LEDs when they are linked to sensors"))
+	}
+	if idx > 7 || idx < 0 {
+		return wrap(fmt.Errorf("invalid led idx %d", idx))
+	}
+	if Debug {
+		fmt.Printf("Set LED state %d - %t\n", idx, state)
+	}
+	if state {
+		return d.setBit(reg_LEDOutputControl, idx)
+	}
+	return d.clearBit(reg_LEDOutputControl, idx)
+}
+
+// Reset issues a soft reset to the device using the reset pin
+// if available.
+func (d *Dev) Reset() (err error) {
+	d.ClearInterrupt()
+	if d != nil && d.opts != nil && d.opts.ResetPin != nil {
+		if Debug {
+			fmt.Println("cap1188: Resetting the device using the reset pin")
+		}
+		if err = d.opts.ResetPin.Out(gpio.Low); err != nil {
+			return err
+		}
+		if err = d.opts.ResetPin.Out(gpio.High); err != nil {
+			return err
+		}
+		time.Sleep(10 * time.Millisecond)
+		if err = d.opts.ResetPin.Out(gpio.Low); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ClearInterrupt resets the interrupt flag
+func (d *Dev) ClearInterrupt() error {
+	// clear the main control bit
+	return d.clearBit(0x0, 0)
+}
+
+// Standby turns off the capacitive touch sensor inputs. The status registers will
+// not be cleared until read. LEDs that are linked to capacitive touch sensor
+// inputs will remain linked and active. Sensor inputs that are no longer
+// sampled will flag a release and then remain in a non-touched state. LEDs that
+// are manually controlled will be unaffected.
+func (d *Dev) Standby() error {
+	return fmt.Errorf("not implemented")
+}
+
+// WakeUp takes the device out of standby or deep sleep mode.
+func (d *Dev) WakeUp() error {
+	return fmt.Errorf("not implemented")
+}
+
+// DeepSleep puts the device in a deep sleep mode. All sensor input scanning is
+// disabled. All LEDs are driven to their programmed non-actuated
+// state and no PWM operations will be done.
+func (d *Dev) DeepSleep() error {
+	return fmt.Errorf("not implemented")
+}
+
+// NewI2C returns a new device that communicates over I²C to cap1188.
+func NewI2C(b i2c.Bus, opts *Opts) (*Dev, error) {
+	addr := uint16(0x28) // default address
+	if opts != nil {
+		switch opts.Address {
+		case 0x28, 0x29, 0x2a, 0x2b, 0x2c:
+			addr = opts.Address
+		case 0x00:
+			// do not do anything
+		default:
+			return nil, wrap(errors.New("given address not supported by device"))
+		}
+	}
+	if Debug {
+		fmt.Printf("cap1188: Connecting via I2C address: %#X\n", addr)
+	}
+	d := &Dev{d: &i2c.Dev{Bus: b, Addr: addr}, opts: opts, isSPI: false}
```

Comment:
Don't keep a pointer to the original options, either copy the whole struct or inline it in the Dev.